File: src/main/java/com/undsf/mcmodmgr/config/CurseForgeConfiguration.java
package com.undsf.mcmodmgr.config;

import com.undsf.mcmodmgr.curseforge.CurseForgeApi;
import okhttp3.OkHttpClient;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class CurseForgeConfiguration {
    @Value("${mmm.curse-forge.api-key}")
    public String apiKey;

    @Autowired
    public OkHttpClient httpClient;

    @Bean("curse-forge-api")
    public CurseForgeApi getCurseForgeApi() {
        return new CurseForgeApi(httpClient, apiKey);
    }
}

File: src/main/kotlin/com/undsf/mcmodmgr/config/CurseForgeConfiguration.kt
package com.undsf.mcmodmgr.config

import com.fasterxml.jackson.databind.ObjectMapper
import com.undsf.mcmodmgr.curseforge.CurseForgeApiClient
import com.undsf.mcmodmgr.util.JsonTemplate
import okhttp3.OkHttpClient
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class CurseForgeConfiguration {
    @Autowired
    lateinit var httpClient: OkHttpClient

    @Autowired
    lateinit var options: CurseForgeOptions

    @Bean("curse-forge-api-client")
    fun getCurseForgeApiClient() : CurseForgeApiClient {
        return CurseForgeApiClient(httpClient, options)
    }
}